R/badge.R

```diff
@@ -138,7 +148,31 @@ use_binder_badge <- function(ref = git_default_branch(), urlpath = NULL) {
 
   invisible(TRUE)
 }
+#' @rdname badges
+#' @export
+use_r_universe_badge <- function() {
+  check_is_package("use_r_universe_badge()")
+  # The r-universe link needs the package name + organization.
 
+  pkg <- project_name()
+  url <- tryCatch(github_url(pkg), error = function(e) NULL)
+  # in order to get organization
+  desc <- proj_desc()
+  urls <- desc$get_urls()
+  dat <- parse_github_remotes(c(urls, url))
+  gh_org <- unique(dat$repo_owner[!is.na(dat$repo_owner)])
+  if (length(gh_org) == 0L) {
+    ui_abort(c(
+      "{.pkg {pkg}} must have a repo URL in DESCRITPION to create a badge.",
+      "Use {.fn usethis::use_badge} if you have a different configuration.",
+      "If {.pkg {pkg}} is on CRAN, you can also see {.url cran.dev/{pkg}}
+       for a redirect to the r-universe homepage."
+    ))
+  }
```

Comment:
This feels like a lot a custom code for a task we do elsewhere, even in other badge functions. Seems like `target_repo_spec()` is relevant.